chainer/training/updater.py

```diff
@@ -0,0 +1,366 @@
+    overriding the :meth:`update` method. In latter case, do not forget to
+    update the iteration counter at each call of this method, because this
+    value is watched by the trainer for deciding when to invoke extensions and
+    when to exit the training loop.
+
+    Args:
+        iterator: Dataset iterator for the training dataset. It can also be a
+            dictionary of iterators. If this is just an iterator, then the
+            iterator is registered by the name ``'main'``.
+        optimizer: Optimizer to update parameters. It can also be a dictionary
+            of optimizers. If this is just an optimizer, then the optimizer is
+            registered by the name ``'main'``.
+        update_func: Update routine. This is a function that takes the updater
+            object as the argument. The default routine uses ``converter``if
+            specified.
+        converter: Converter function to build input arrays. Each batch
+            extracted by the main iterator is split equaly between the
+            devices and then passed with corresponding ``device`` option to
+            this function. If it is omitted,
+            :func:`~chainer.dataset.concat_examples` is used.
+        models: Dictionary of models. The main model should be the same model
+            attached to the ``'main'`` optimizer.
+        devices: Dictionary of devices to which the training data is sent. The
+            devices should be arranged in a dictionary with the same structure
+            as ``models``.
+
+    Attributes:
+        iteration: Current number of completed updates.
+
+    """
+    def __init__(self, iterator, optimizer, update_func=None, converter=None,
+                 models=None, devices=None):
+        super(ParallelUpdater, self).__init__(
+            iterator=iterator,
+            optimizer=optimizer,
+            converter=converter
+        )
+        assert models is not None or devices is not None, \
+            "Either 'models' or 'devices' must be defined."
+
+        if models is None:
+            names = list(six.iterkeys(devices))
+            try:
+                names.remove('main')
+            except ValueError:
+                raise KeyError("'devices' must contain a 'main' key.")
+            models = {'main': optimizer.target}
+            for name in names:
+                models[name] = models['main'].copy()
+
+        if devices is None:
+            devices = dict((name, i) for i, name in enumerate(names))
+
+        for name, model in models.items():
+            model.to_gpu(devices[name])
+
+        self._models = models
+
+        self._update_func = update_func or _parallel_update(
+            self, converter, models, devices)
+
+    def connect_trainer(self, trainer):
+        """Connects the updater to the trainer that will call it.
+
+        The typical usage of this method is to register additional links to the
```

Comment:
ditto (use the reference to the document of `StandardUpdate` instead of copying it).